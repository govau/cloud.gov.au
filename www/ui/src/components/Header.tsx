import * as React from "react";
import styled, { css } from "../styled-components";
import { Flex, Box } from "grid-styled";
import { responsiveStyle } from "styled-system";

import * as coaLogo from "./coa.svg";
import BaseLink, { LinkProps, RouterLinkProps } from "./Link";
import Container from "./Container";
import HeaderLinks from "./HeaderLinks";

const mediaDisplay = responsiveStyle({
  cssProperty: "display"
});

const COABox = styled(Box)`
  ${mediaDisplay};
`;

const COALogo = styled.img`
  height: 5rem;
  padding-right: 1rem;
  ${({ theme }) =>
    css`
      border-right: 1px solid ${theme.coaDividerColor};
    `};
`;

const UnstyledBrandLink: React.StatelessComponent<
  Partial<LinkProps & RouterLinkProps>
> = ({ children, ...rest }) => <BaseLink {...rest}>{children}</BaseLink>;

const BrandLink = styled(UnstyledBrandLink)`
  display: block;
  border-bottom: 0;
`;

const Brand = styled.div`
  margin: 0;
  font-weight: 500;
  font-size: 1.5rem;
`;

const StyledHeader = styled.header`
  ${({ theme }) =>
    css`
      background-color: ${theme.headerColor};
    `};
  color: #ffffff;
`;

interface Props {}

const Header: React.StatelessComponent<Props> = () => (
  <StyledHeader>
    <Container>
      <Flex wrap={true} pt={3} pb={3} align="center">
        <COABox pr={2} display={["none", "block"]}>
          <a href="/">
            <COALogo
              src={coaLogo}
              alt="The Australian Government Coat of Arms"
            />
          </a>
        </COABox>
        <Box>
          <BrandLink to="/">
            <Brand>cloud.gov.au</Brand>
          </BrandLink>
        </Box>
        <Box w={[1, "auto"]} mt={[2, 0]} ml={[0, "auto"]}>
          <HeaderLinks />
        </Box>
      </Flex>
    </Container>
  </StyledHeader>
);

export default Header;
