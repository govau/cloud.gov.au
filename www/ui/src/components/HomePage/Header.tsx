import * as React from "react";
import styled, { css } from "../../styled-components";
import { Flex, Box } from "grid-styled";

import Container from "../Container";

const Heading = styled.h1`
  margin-top: 0;
  margin-bottom: 0;
  font-size: 4rem;
  font-weight: 700;
`;

const StyledHeader = styled.div`
  ${({ theme }) =>
    css`
      background-color: ${theme.headerColor};
    `};
  color: #ffffff;
`;

interface Props {}

const Header: React.StatelessComponent<Props> = () => (
  <StyledHeader>
    <Container>
      <Flex wrap={true} pt={2} pb={3}>
        <Box>
          <Heading>Cloud</Heading>
        </Box>
      </Flex>
    </Container>
  </StyledHeader>
);

export default Header;
