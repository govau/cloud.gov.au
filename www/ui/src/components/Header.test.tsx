import * as React from "react";
import { MemoryRouter } from "react-router";
import * as renderer from "react-test-renderer";

import Header from "./Header";

it("renders without crashing", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
