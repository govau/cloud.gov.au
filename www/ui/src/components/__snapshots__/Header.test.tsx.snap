// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c9 {
  -webkit-transition: 0.25s;
  transition: 0.25s;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 1px;
}

.c5 {
  -webkit-transition: 0.25s;
  transition: 0.25s;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: solid 1px;
}

.c1 {
  max-width: 1024px;
}

.c7 {
  margin: 0;
  padding: 0;
  list-style: none;
}

.c8 {
  display: block;
  margin: 8px;
}

.c2 {
  box-sizing: border-box;
  padding-right: 16px;
  display: none;
}

.c3 {
  height: 5rem;
  padding-right: 1rem;
  border-right: 1px solid;
}

.c4 {
  display: block;
  border-bottom: 0;
}

.c6 {
  margin: 0;
  font-weight: 500;
  font-size: 1.5rem;
}

.c0 {
  color: #ffffff;
}

@media screen and (min-width:40em) {
  .c8 {
    display: inline-block;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    display: block;
  }
}

<header
  className="c0"
>
  <div
    className="c1 bNqJOV"
    is={null}
  >
    <div
      className="fzUmoZ NcCKj"
      is={null}
    >
      <div
        className="c2"
        display={
          Array [
            "none",
            "block",
          ]
        }
        is={null}
      >
        <a
          href="/"
        >
          <img
            alt="The Australian Government Coat of Arms"
            className="c3"
            src={
              Object {
                "0": "c",
                "1": "o",
                "2": "a",
                "3": ".",
                "4": "s",
                "5": "v",
                "6": "g",
                "default": "coa.svg",
              }
            }
          />
        </a>
      </div>
      <div
        className="iHZvIS"
        is={null}
      >
        <a
          className="c4 c5"
          href="/"
          onClick={[Function]}
        >
          <div
            className="c6"
          >
            cloud.gov.au
          </div>
        </a>
      </div>
      <div
        className="fvqBgo"
        is={null}
      >
        <ul
          className="c7"
        >
          <li
            className="c8"
            display={
              Array [
                "block",
                "inline-block",
              ]
            }
            m={
              Array [
                1,
              ]
            }
          >
            <a
              className="c5"
              href="/insights/"
              onClick={[Function]}
            >
              Insights
            </a>
          </li>
          <li
            className="c8"
            display={
              Array [
                "block",
                "inline-block",
              ]
            }
            m={
              Array [
                1,
              ]
            }
          >
            <a
              className="c5"
              href="/code/"
              onClick={[Function]}
            >
              Code
            </a>
          </li>
          <li
            className="c8"
            display={
              Array [
                "block",
                "inline-block",
              ]
            }
            m={
              Array [
                1,
              ]
            }
          >
            <a
              className="c9"
              href="https://docs.cloud.gov.au/"
            >
              Docs
            </a>
          </li>
          <li
            className="c8"
            display={
              Array [
                "block",
                "inline-block",
              ]
            }
            m={
              Array [
                1,
              ]
            }
          >
            <a
              className="c9"
              href="https://support.cloud.gov.au/"
            >
              Support
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</header>
`;
